Building boundless-market sometimes fails "unexpected end of input"

This build failure has been seen on some machines consistently when building via docker or compiling directly on the machine.

```
2025-11-12T21:54:21.9912220Z #36 80.05   assessor-guest:    Compiling risc0-zkvm v3.0.3
2025-11-12T21:54:21.9913986Z #36 80.05   assessor-guest: error: unexpected end of input, expected one of: `constructor`, `function`, `fallback`, `receive`, `modifier`, `struct`, `event`, `error`, `abstract`, `contract`, `interface`, `library`, `enum`, `type`, `pragma`, `import`, `using`, parentheses, `tuple`, `function`, `mapping`, identifier
2025-11-12T21:54:21.9916637Z #36 80.05   assessor-guest:    --> /src/target/riscv-guest/guest-assessor/assessor-guest/riscv32im-risc0-zkvm-elf/release/build/boundless-market-014af43ab13e9713/out/boundless_market_generated.rs:3:13
2025-11-12T21:54:21.9918003Z #36 80.05   assessor-guest:     |
2025-11-12T21:54:21.9918546Z #36 80.05   assessor-guest: 3   | /             alloy_sol_types::sol! {
2025-11-12T21:54:21.9919175Z #36 80.05   assessor-guest: 4   | |             #![sol(all_derives)]
2025-11-12T21:54:21.9919997Z #36 80.05   assessor-guest: 5   | |             #![sol(extra_derives(serde::Serialize, serde::Deserialize))]
2025-11-12T21:54:21.9920655Z #36 80.05   assessor-guest: ...   |
2025-11-12T21:54:21.9921053Z #36 80.05   assessor-guest: 884 | | }
2025-11-12T21:54:21.9921464Z #36 80.05   assessor-guest:     | |_^
2025-11-12T21:54:21.9921866Z #36 80.05   assessor-guest:     |
2025-11-12T21:54:21.9922806Z #36 80.05   assessor-guest:     = note: this error originates in the macro `alloy_sol_types::sol` (in Nightly builds, run with -Z macro-backtrace for more info)
2025-11-12T21:54:21.9923771Z #36 80.05   assessor-guest: 
2025-11-12T21:54:21.9929422Z #36 80.05   assessor-guest: error[E0432]: unresolved imports `boundless_market_contract::AssessorCallback`, `boundless_market_contract::AssessorCommitment`, `boundless_market_contract::AssessorJournal`, `boundless_market_contract::AssessorReceipt`, `boundless_market_contract::Callback`, `boundless_market_contract::Fulfillment`, `boundless_market_contract::FulfillmentContext`, `boundless_market_contract::FulfillmentDataImageIdAndJournal`, `boundless_market_contract::FulfillmentDataType`, `boundless_market_contract::IBoundlessMarket`, `boundless_market_contract::Input`, `boundless_market_contract::InputType`, `boundless_market_contract::LockRequest`, `boundless_market_contract::Offer`, `boundless_market_contract::Predicate`, `boundless_market_contract::PredicateType`, `boundless_market_contract::ProofRequest`, `boundless_market_contract::RequestLock`, `boundless_market_contract::Requirements`, `boundless_market_contract::Selector`
2025-11-12T21:54:21.9935301Z #36 80.05   assessor-guest:   --> /src/crates/boundless-market/src/contracts/mod.rs:63:5
2025-11-12T21:54:21.9935911Z #36 80.05   assessor-guest:    |
2025-11-12T21:54:21.9936763Z #36 80.05   assessor-guest: 63 |     AssessorCallback, AssessorCommitment, AssessorJournal, AssessorReceipt, Callback, Fulfillment,
2025-11-12T21:54:21.9937874Z #36 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^  ^^^^^^^^^^^ no `Fulfillment` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9938686Z #36 80.05   assessor-guest:    |     |                 |                   |                |                |
2025-11-12T21:54:21.9939488Z #36 80.05   assessor-guest:    |     |                 |                   |                |                no `Callback` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9940522Z #36 80.05   assessor-guest:    |     |                 |                   |                no `AssessorReceipt` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9941475Z #36 80.05   assessor-guest:    |     |                 |                   no `AssessorJournal` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9942420Z #36 80.05   assessor-guest:    |     |                 no `AssessorCommitment` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9943356Z #36 80.05   assessor-guest:    |     no `AssessorCallback` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9944541Z #36 80.05   assessor-guest: 64 |     FulfillmentContext, FulfillmentDataImageIdAndJournal, FulfillmentDataType, IBoundlessMarket,
2025-11-12T21:54:21.9945757Z #36 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^ no `IBoundlessMarket` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9946581Z #36 80.05   assessor-guest:    |     |                   |                                 |
2025-11-12T21:54:21.9947431Z #36 80.05   assessor-guest:    |     |                   |                                 no `FulfillmentDataType` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9948530Z #36 80.05   assessor-guest:    |     |                   no `FulfillmentDataImageIdAndJournal` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9949582Z #36 80.05   assessor-guest:    |     no `FulfillmentContext` in `contracts::boundless_market_contract`
2025-11-12T21:54:21.9950645Z #36 80.05   assessor-guest: 65 |     Input as RequestInput, InputType as RequestInputType, LockRequest, Offer,
2025-11-12T21:54:21.9951523Z #36 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^
2025-11-12T21:54:21.9952486Z #36 80.05   assessor-guest: 66 |     Predicate as RequestPredicate, PredicateType, ProofRequest, RequestLock, Requirements,
2025-11-12T21:54:21.9953453Z #36 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^^^^
2025-11-12T21:54:21.9954117Z #36 80.05   assessor-guest: 67 |     Selector as AssessorSelector,
2025-11-12T21:54:21.9954717Z #36 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-12T21:54:21.9955202Z #36 80.05   assessor-guest: 
2025-11-12T21:54:21.9955830Z #36 80.05   assessor-guest: For more information about this error, try `rustc --explain E0432`.
2025-11-12T21:54:21.9956800Z #36 80.05   assessor-guest: error: could not compile `boundless-market` (lib) due to 2 previous errors
2025-11-12T21:54:21.9957717Z #36 80.05   assessor-guest: warning: build failed, waiting for other jobs to finish...
2025-11-12T21:54:21.9958668Z #36 80.05 warning: build failed, waiting for other jobs to finish...
2025-11-12T21:59:53.4694915Z #36 ERROR: process "/bin/bash -c cargo build --release --bin broker &&     cp /src/target/release/broker /src/broker" did not complete successfully: exit code: 101
2025-11-12T21:59:53.4882961Z ------
2025-11-12T21:59:53.4883710Z  > [builder 14/14] RUN cargo build --release --bin broker &&     cp /src/target/release/broker /src/broker:
2025-11-12T21:59:53.4884688Z 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^
2025-11-12T21:59:53.4885557Z 80.05   assessor-guest: 66 |     Predicate as RequestPredicate, PredicateType, ProofRequest, RequestLock, Requirements,
2025-11-12T21:59:53.4886438Z 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^  ^^^^^^^^^^^^
2025-11-12T21:59:53.4887048Z 80.05   assessor-guest: 67 |     Selector as AssessorSelector,
2025-11-12T21:59:53.4887596Z 80.05   assessor-guest:    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-12T21:59:53.4888053Z 80.05   assessor-guest: 
2025-11-12T21:59:53.4888625Z 80.05   assessor-guest: For more information about this error, try `rustc --explain E0432`.
2025-11-12T21:59:53.4889522Z 80.05   assessor-guest: error: could not compile `boundless-market` (lib) due to 2 previous errors
2025-11-12T21:59:53.4890540Z 80.05   assessor-guest: warning: build failed, waiting for other jobs to finish...
2025-11-12T21:59:53.4891222Z 80.05 warning: build failed, waiting for other jobs to finish...
2025-11-12T21:59:53.4891681Z ------
2025-11-12T21:59:53.4895586Z broker.dockerfile:64
2025-11-12T21:59:53.4896069Z --------------------
2025-11-12T21:59:53.4896452Z   63 |     
2025-11-12T21:59:53.4896880Z   64 | >>> RUN cargo build --release --bin broker && \
2025-11-12T21:59:53.4897501Z   65 | >>>     cp /src/target/release/broker /src/broker
2025-11-12T21:59:53.4898806Z   66 |     
2025-11-12T21:59:53.4899159Z --------------------
2025-11-12T21:59:53.4900593Z ERROR: failed to solve: process "/bin/bash -c cargo build --release --bin broker &&     cp /src/target/release/broker /src/broker" did not complete successfully: exit code: 101
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 24.04.3 LTS
Release:	24.04
Codename:	noble
$ rustc --version
rustc 1.88.0 (6b00bc388 2025-06-23)
$ forge --version
forge Version: 1.3.5-v1.3.5
Commit SHA: 9979a41b5daa5da1572d973d7ac5a3dd2afc0221
Build Timestamp: 2025-09-08T15:20:01.044260027Z (1757344801)
Build Profile: maxperf
```

While the cause remains unknown this change does resolve the issue:

```git
diff --git a/contracts/src/types/FulfillmentContext.sol b/contracts/src/types/FulfillmentContext.sol
index f6a06caf..32d8c196 100644
--- a/contracts/src/types/FulfillmentContext.sol
+++ b/contracts/src/types/FulfillmentContext.sol
@@ -18,9 +18,6 @@ struct FulfillmentContext {
     uint96 price;
 }

-/// @title FulfillmentContextLibrary
-/// @notice Library for packing, unpacking, and storing FulfillmentContext structs
-/// @dev Uses bit manipulation to pack all fields into a single uint256 for transient storage
 library FulfillmentContextLibrary {
     uint256 private constant VALID_MASK = 1 << 127;
     uint256 private constant EXPIRED_MASK = 1 << 126;
diff --git a/crates/boundless-market/src/contracts/artifacts/FulfillmentContext.sol b/crates/boundless-market/src/contracts/artifacts/FulfillmentContext.sol
index f6a06caf..32d8c196 100644
```

--- a/contracts/scripts/BoundlessScript.s.sol
+++ b/contracts/scripts/BoundlessScript.s.sol
@@ -10,7 +10,6 @@ import {Script, console2} from "forge-std/Script.sol";
 import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
 import {ConfigLoader, DeploymentConfig} from "./Config.s.sol";
 
-/// @notice Shared library for Boundless deployment and management scripts
 library BoundlessScript {
     /// @notice Validates that an address value is not zero, with descriptive error message
     function requireLib(address value, string memory label) internal pure returns (address) {

--- a/contracts/src/config/VerifierConfig.sol
+++ b/contracts/src/config/VerifierConfig.sol
@@ -71,8 +71,6 @@ library DeploymentLib {
     }
 }
 
-/// @notice Loader for the deployment config from a given deployment.toml file.
-/// @dev This library uses Forge cheat code and can only be used in Forge script or test environments.
 library ConfigLoader {
     /// Reference the vm address without needing to inherit from Script.
     Vm private constant VM = Vm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
@@ -117,8 +115,6 @@ library ConfigLoader {
     }
 }
 
-/// @notice Parser for the deployment config given a TOML string.
-/// @dev This library uses Forge cheat code and can only be used in Forge script or test environments.
 library ConfigParser {
     using SafeCast for uint256;
 

--- a/contracts/src/types/FulfillmentContext.sol
+++ b/contracts/src/types/FulfillmentContext.sol
@@ -18,9 +18,6 @@ struct FulfillmentContext {
     uint96 price;
 }
 
-/// @title FulfillmentContextLibrary
-/// @notice Library for packing, unpacking, and storing FulfillmentContext structs
-/// @dev Uses bit manipulation to pack all fields into a single uint256 for transient storage
 library FulfillmentContextLibrary {
     uint256 private constant VALID_MASK = 1 << 127;
     uint256 private constant EXPIRED_MASK = 1 << 126;

--- a/crates/boundless-market/src/contracts/artifacts/FulfillmentContext.sol
+++ b/crates/boundless-market/src/contracts/artifacts/FulfillmentContext.sol
@@ -18,9 +18,6 @@ struct FulfillmentContext {
     uint96 price;
 }
 
-/// @title FulfillmentContextLibrary
-/// @notice Library for packing, unpacking, and storing FulfillmentContext structs
-/// @dev Uses bit manipulation to pack all fields into a single uint256 for transient storage
 library FulfillmentContextLibrary {
     uint256 private constant VALID_MASK = 1 << 127;
     uint256 private constant EXPIRED_MASK = 1 << 126;

